syft/pkg/cataloger/java/parse_jvm_release.go

```diff
@@ -0,0 +1,367 @@
+// OS_ARCH="aarch64"
+// OS_NAME="Linux"
+// SOURCE=".:git:13710926b798"
+// BUILD_SOURCE="git:1271f10a26c47e1489a814dd2731f936a588d621"
+// BUILD_SOURCE_REPO="https://github.com/adoptium/temurin-build.git"
+// SOURCE_REPO="https://github.com/adoptium/jdk21u.git"
+// FULL_VERSION="21.0.4+7-LTS"
+// SEMANTIC_VERSION="21.0.4+7"
+// BUILD_INFO="OS: Linux Version: 5.4.0-150-generic"
+// JVM_VARIANT="Hotspot"
+// JVM_VERSION="21.0.4+7-LTS"
+// IMAGE_TYPE="JDK"
+//
+// In terms of the temurin flavor, these are controlled by:
+// - config: https://github.com/adoptium/temurin-build/blob/v2023.01.03/sbin/common/config_init.sh
+// - build script: https://github.com/adoptium/temurin-build/blob/v2023.01.03/sbin/build.sh#L1584-L1796
+
+func parseJVMRelease(_ context.Context, resolver file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	ri, err := parseJvmReleaseInfo(reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to parse JVM release info %q: %w", reader.Path(), err)
+	}
+
+	if ri == nil {
+		// TODO: known-unknown: expected JDK installation package
+		return nil, nil, nil
+	}
+
+	installDir := path.Dir(reader.Path())
+
+	version := jvmVersion(ri)
+
+	licenses := jvmLicenses(resolver, ri)
+
+	locations := file.NewLocationSet(reader.Location)
+
+	for _, lic := range licenses.ToSlice() {
+		locations.Add(lic.Locations.ToSlice()...)
+	}
+
+	p := pkg.Package{
+		Name:      jvmPackageNameFromPath(installDir),
+		Locations: locations,
+		Version:   version,
+		CPEs:      jvmCpes(ri.Implementor, version),
+		PURL:      jvmPurl(version),
+		Licenses:  licenses,
+		Type:      pkg.BinaryPkg,
+		Metadata: pkg.JvmInstallation{
+			ReleaseInfo: *ri,
+			Files:       findJVMFiles(resolver, installDir),
+		},
+	}
+	p.SetID()
+
+	return []pkg.Package{p}, nil, nil
+}
+
+func jvmLicenses(_ file.Resolver, _ *pkg.JvmReleaseInfo) pkg.LicenseSet {
+	// TODO: get this from the dir(<RELEASE>)/legal/**/LICENSE files when we start cataloging license content
+	// see https://github.com/anchore/syft/issues/656
+	return pkg.NewLicenseSet()
+}
+
+func findJVMFiles(resolver file.Resolver, installDir string) []string {
+	ownedLocations, err := resolver.FilesByGlob(installDir + "/**")
+	if err != nil {
+		// TODO: known-unknowns
+		log.WithFields("path", installDir, "error", err).Trace("unable to find installed JVM files")
+	}
+
+	var results []string
+	for _, loc := range ownedLocations {
+		results = append(results, loc.Path())
+	}
+
+	sort.Strings(results)
+
+	return results
+}
+
+func jvmPurl(version string) string {
+	pURL := packageurl.NewPackageURL(
+		packageurl.TypeGeneric,
+		"oracle",
+		"openjdk",
+		version,
+		nil,
+		"")
+	return pURL.ToString()
+}
+
+func jvmCpes(implementor, version string) []cpe.CPE {
```

Comment:
just calling this out, I think this still needs some work, we should be raising up `oracle:jdk` vs `oracle:openjdk` vs `ibm:jdk` etc...